### PR TITLE
feat: create social network navigation alike on browse threads

### PR DESCRIPTION
**Pull request by TopETH**
## Summary by CodeRabbit

- **New Features**
	- Enhanced the user menu with asynchronous sign out and automatic page refresh upon signing out.
	- Improved accordion components with new functionalities for better URL handling and state management.

- **Bug Fixes**
	- Updated slug generation logic to replace '&' with 'n', ensuring more accurate and readable URL slugs.

### CODE REVIEW

**Review by gaboesquivel — CHANGES_REQUESTED** [review]
@TopETH  merge the latest changes, notice that we now `lib/url` and` lib/username`  for slug and url encoding

**Comment by TopETH** [comment]
> @TopETH merge the latest changes, notice that we now `lib/url` and` lib/username` for slug and url encoding

Done.

File: apps/masterbots.ai/components/shared/thread-double-accordion.tsx [hunk 1]
```diff
@@ -10,28 +10,75 @@ import {
...
+  }, [])
+
+  const toggleAccordion = (v: string[]) => {
+    const isOpen = Boolean(v[0] === `pair-${thread.threadId}`)
+    setState({ isOpen, value: v })
+    if (isOpen) {
+      if (!activeThread) initialUrl = location.href
+      setActiveThread(thread)
+      const dir = chat
+        ? 'c/' +
```
Comment (gaboesquivel) [inline apps/masterbots.ai/components/shared/thread-double-accordion.tsx:55]: `import {toSlug} from 'lib/url'`

const dir = chat ? `/c/${toSlug(thread.chatbot.name)}` : `toSlug(thread.chatbot.categories[0].category.name)`

I assigned this to you as well as it's related https://github.com/bitcashorg/masterbots/issues/163

File: apps/masterbots.ai/lib/url.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ export const SlugSchema: ZodSchema<string> = z.string()
 export const toSlug = (username: string, separator = "_"): string => {
   return username
     .toLowerCase()
-    .replace(/&/g, '_')
-    .replace(/ & /g, '_')
+    .replace(/&/g, 'n')
```
Comment (gaboesquivel) [inline apps/masterbots.ai/lib/url.ts:12]: please rollback this change on toSlug funciton. update the link on everywhere to use the same slug format without modification https://github.com/bitcashorg/masterbots/issues/163

**Review by gaboesquivel — CHANGES_REQUESTED** [review]

File: apps/masterbots.ai/components/shared/thread-accordion.tsx [hunk 3]
```diff
@@ -32,54 +32,54 @@ export function ThreadAccordion({
...
 
-  // update url when dialog opens and closes
-  useEffect(() => {
-    const initialUrl = location.href
-    const dir =
-      `c/${ 
-      toSlug(
-        chat ? thread.chatbot.name : thread.chatbot.categories[0].category.name
-      )}`
+  // // update url when dialog opens and closes
```
Comment (gaboesquivel) [inline apps/masterbots.ai/components/shared/thread-accordion.tsx:35]: you cannot disable this functionality of updating the url when the thread is expanded

File: apps/masterbots.ai/components/shared/thread-double-accordion.tsx [hunk 4]
```diff
@@ -10,28 +10,69 @@ import {
...
+    if (activeThread === thread || !state.isOpen) return
+    setState({ isOpen: false, value: [] })
+  }, [activeThread])
+
+  useEffect(() => {
+    if (activeThread !== thread || state.isOpen) return
+    setState({ isOpen: true, value: [`pair-${thread.threadId}`] })
+  }, [])
+
+  const toggleAccordion = (v: string[]) => {
```
Comment (gaboesquivel) [inline apps/masterbots.ai/components/shared/thread-double-accordion.tsx:49]: this logic goes in ThreadAccordion as is this also needs to work on the ThreadDialog used for chat and cmdk

File: apps/masterbots.ai/components/shared/thread-accordion.tsx [hunk 5]
```diff
@@ -32,54 +32,54 @@ export function ThreadAccordion({
...
+      className="w-full flex flex-col gap-3"
       defaultValue={['pair-0', 'pair-1', 'pair-2']}
       type="multiple"
     >
       {pairs.map((p, key) => {
         return (
-          <AccordionItem key={key} value={`pair-${key}`}>
-            {showHeading ? (
+          <AccordionItem value={`pair-${key}`} key={key}>
+            {key ? (
```
Comment (gaboesquivel) [inline apps/masterbots.ai/components/shared/thread-accordion.tsx:68]: this is hard to understand, theres  is no need to tho have 2 AccordionTrigger elements, simply change what goes inside so its easier to read and follow.

If the ThreadAccordion is displayed inside ThreadDialog or ThreadDoubleAccordion then key is not display

If ThreadAccordion is used in the Thread landing page we need to display Heading.

**Review by gaboesquivel — CHANGES_REQUESTED** [review]

**Comment by gaboesquivel** [comment]
ThreadAccordion is used in many different places, make sure it works correctly on all of the ,   updating the url needs to happen both we you open ThreadDialog on chat and when you expand DoubleThreadDialog in browse,  you also need to make the chat landing and browse landing pages work. that 4 different places

File: apps/masterbots.ai/components/shared/thread-accordion.tsx (hunk 3 above)
Comment (TopETH) [reply apps/masterbots.ai/components/shared/thread-accordion.tsx:35]: @gaboesquivel Did you check it on the site?
It is not working correctly. So we should disable it and when I fix the chat features, I am gonna fix it as well.

File: apps/masterbots.ai/components/shared/thread-double-accordion.tsx (hunk 4 above)
Comment (TopETH) [reply apps/masterbots.ai/components/shared/thread-double-accordion.tsx:49]: I need to fix this issue while fixing the chat feature.

**Pull request closed without merging**